### PR TITLE
emoji_test: update to use testify

### LINKED ISSUE #12861: Migrate tests from "api4/emoji_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `api4/emoji_test.go` file.

The expected way to migrate this is to go to the `api4/emoji_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require) if a failure should stop the execution of the test, or [the `assert` package](https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19624

### PR DESCRIPTION
**Pull request by drekar**
#### Summary
Update `api4/emoji_test.go` to use `testify`.

#### Ticket Link
Fixes #12861 
[MM-19624](https://mattermost.atlassian.net/browse/MM-19624)

### CODE REVIEW

File: api4/emoji_test.go [hunk 1]
```diff
@@ -506,15 +476,10 @@ func TestGetEmojiImage(t *testing.T) {
...
 	emojiImage, resp := Client.GetEmojiImage(emoji1.Id)
 	CheckNoError(t, resp)
-	if len(emojiImage) <= 0 {
-		t.Fatal("should return the image")
-	}
+	require.Greater(t, len(emojiImage), 0, "should return the image")
+
 	_, imageType, err := image.DecodeConfig(bytes.NewReader(emojiImage))
-	if err != nil {
-		t.Fatalf("unable to identify received image: %v", err.Error())
```
Comment (iomodo) [inline api4/emoji_test.go]: `require.NoError` missing

File: api4/emoji_test.go [hunk 2]
```diff
@@ -245,16 +227,12 @@ func TestGetEmojiList(t *testing.T) {
 	listEmoji, resp = Client.GetEmojiList(0, 1)
 	CheckNoError(t, resp)
 
-	if len(listEmoji) != 1 {
-		t.Fatal("should only return 1")
-	}
+	require.Equal(t, len(listEmoji), 1, "should only return 1")
```
[the remark this answers is not in the document]
Comment (drekar) [reply api4/emoji_test.go]: Thank you, I missed that one in the godocs!

File: api4/emoji_test.go (hunk 1 above)
Comment (drekar) [reply api4/emoji_test.go]: Thanks!

**Comment by drekar** [comment]
> One other thing, there are some `assert.True` and `assert.Equal` in the code. I'm 0/5 whether we should change them to `require` or not.

Thanks @iomodo ; I am also a fan of replacing assert statements with require.  The moment than an expected condition is not met, the underlying state machine is in an unknown position, and thus subsequent tests can not be trusted.  "Fail Fast and Fail Often", or so the aphorism goes.
However, I am receiving suggestions to go the other way on other PRs, which is likely a pragmatic approach of displaying as many problems in one test-run as possible to save time at the expense of strict correctness, which is a position I fully understand.  So for now, I suppose I'll going to leave things as-is due to ambiguity :)

**Pull request merged**